src/components/ActionsBuilder/SupportedActions/RepMint/RepMintEditor.tsx

```diff
@@ -31,29 +31,26 @@ export const Mint: React.FC<ActionEditorProps> = ({
 }) => {
   const { t } = useTranslation();
   const [repAmount, setRepAmount] = useState<string>('0');
-  const { parsedData } = useTotalSupply({ decodedCall });
+  const { data } = useTotalSupply({ decodedCall });
   const { tokenData } = useTokenData();
-  const [recipient, setRecipient] = useState<string>(parsedData?.toAddress);
+  const [recipient, setRecipient] = useState<string>(data?.toAddress);
   const totalSupply = useBigNumberToNumber(tokenData?.totalSupply, 18);
-  const { imageUrl } = useENSAvatar(parsedData?.toAddress, MAINNET_ID);
+  const { imageUrl } = useENSAvatar(data?.toAddress, MAINNET_ID);
 
   const { control, handleSubmit, setValue } = useForm<RepMintFormValues>({
     resolver: validateRepMint,
     context: { t },
   });
 
   useEffect(() => {
-    const initialRepAmount = ethers.utils.formatEther(
-      parsedData?.amount?.toString()
-    );
-    if (initialRepAmount) {
-      setRepAmount(initialRepAmount);
+    if (data?.amount) {
+      setRepAmount((data?.amount).toString());
       setValue(
         'repPercent',
-        String((Number(initialRepAmount) * 100) / totalSupply || 0)
+        String((Number(data?.amount) * 100) / totalSupply || 0)
       );
     }
-  }, []); //eslint-disable-line
+  }, [data]); //eslint-disable-line

```

Comment:
If `data.amount` is primitive, for sure it will be better to put only`[data.amount]` into the deeps array. Feel free to omit this